test/hotspot/jtreg/compiler/lib/ir_framework/driver/irmatching/parser/VMInfo.java

```diff
@@ -0,0 +1,119 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.lib.ir_framework.driver.irmatching.parser;
+
+import compiler.lib.ir_framework.TestFramework;
+import compiler.lib.ir_framework.shared.TestFrameworkException;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class stores the key value mapping from the VMInfo.
+ *
+ * @see IREncodingParser
+ */
+public class VMInfo {
+    /**
+     * Stores the key-value mapping.
+     */
+    private final Map<String, String> keyValueMap;
+
+    private static final Pattern CPU_SKYLAKE_PATTERN =
+            Pattern.compile("family 6 model 85 stepping (\\d) ");
+
+    public VMInfo(Map<String, String> map) {
+        this.keyValueMap = map;
+
+        TestFramework.check(isKey("cpuFeatures"),   "VMInfo does not contain cpuFeatures");
+        TestFramework.check(isKey("MaxVectorSize"), "VMInfo does not contain MaxVectorSize");
+        TestFramework.check(isKey("MaxVectorSizeIsDefault"), "VMInfo does not contain MaxVectorSizeIsDefault");
+        TestFramework.check(isKey("LoopMaxUnroll"), "VMInfo does not contain LoopMaxUnroll");
+        TestFramework.check(isKey("UseAVX"), "VMInfo does not contain UseAVX");
+        TestFramework.check(isKey("UseAVXIsDefault"), "VMInfo does not contain UseAVXIsDefault");
+        System.err.println("--- VMInfo from Test VM ---");
+        System.err.println("cpuFeatures:   " + getStringValue("cpuFeatures"));
+        System.err.println("MaxVectorSize: " + getLongValue("MaxVectorSize"));
+        System.err.println("MaxVectorSizeIsDefault: " + getLongValue("MaxVectorSizeIsDefault"));
+        System.err.println("LoopMaxUnroll: " + getLongValue("LoopMaxUnroll"));
+        System.err.println("UseAVX: " + getLongValue("UseAVX"));
+        System.err.println("UseAVXIsDefault: " + getLongValue("UseAVXIsDefault"));
+        if (isDefaultCascadeLake()) {
+            System.err.println(" -> You are on default Cascade Lake");
+            System.err.println("   -> SuperWord expected to run with 32 byte, not 64 byte, VectorAPI expected to use 64 byte");
+            System.err.println("   -> \"canTrustVectorSize == false\", some vector node IR rules are made weaker.");
+        }
```

Comment:
Is this a leftover from debugging? If you want to print this information for debugging purposes, I suggest to move this code to `VMInfoParser` and additionally guard it with `VERBOSE || PRINT_IR_ENCODING`. The name `PRINT_IR_ENCODING` is not completely correct here but we might want to clean this up separately at some other point in time.

You can keep the verification of calling the `get*()` methods here, though.

Reply:
I can just remove it. It is not necessary any more I think.